cmd/keytransparency-client/cmd/root.go

```diff
@@ -217,22 +219,32 @@ func dial(ktURL, caFile, clientSecretFile string, serviceKeyFile string) (*grpc.
 		opts = append(opts, grpc.WithTransportCredentials(creds))
 	}
 
-	// Add client credentials otherwise add service credentials. Client
-	// credentials take priority over service credentials. Only one of the
-	// two should exist in an RPC call.
-	if clientSecretFile != "" {
-		creds, err := getCreds(clientSecretFile)
-		if err != nil {
-			return nil, err
-		}
-		opts = append(opts, grpc.WithPerRPCCredentials(creds))
-	} else if serviceKeyFile != "" {
-		creds, err := getServiceCreds(serviceKeyFile)
-		if err != nil {
-			return nil, err
+	// Add authentication information for the grpc.
+	fakeUserId := viper.GetString("fake-auth-userid")
+	if fakeUserId != "" { // Using fake authentication.
+		// Fake authentication information are added in cmd/keytransparency-client/cmd/post.go
+		// TODO(amarcedone) Consider having the GetClient method (or this method) initialize
+		// the context so that we can set the metadata parameters for the fake authentication
+		// here rather than in post.go.
+	} else {
```

Comment:
This set of if/else's could be replaced with a `switch`, one case for each type of authentication.